### PR TITLE
GH-3146: use try-with-resources (via Netbeans Inspect and Transform)

### PR DESCRIPTION
**Pull request by barthanssens**
Signed-off-by:Bart Hanssens <bart.hanssens@bosa.fgov.be>

GitHub issue resolved: #3146 

Briefly describe the changes proposed in this PR:

- use try-with-resources instead of try-finally

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

File: core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java [hunk 1]
```diff
@@ -195,10 +195,8 @@ public void setHttpClient(HttpClient httpClient) {
 
 	@Override
 	public void close() throws RepositoryException {
-		try {
+		try (client) {
```
Comment (jeenbroekstra) [inline core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java:198]: I've not seen try-with-resources used in this way before (autoclosing something that was defined out of scope of the actual try block). Could we do a quick smoke test that that actually works as expected?

**Review by jeenbroekstra — COMMENTED** [review]
Mostly looks good to me, nice bit of (auto) refactoring. One question about a bit of an odd pattern I noticed.

File: core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java (hunk 1 above)
Comment (barthanssens) [reply core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java:198]: TBH I haven't tested all of them, and I'm thinking to roll back those (auto-generated) changes.
I assume it (mostly ?) works, but at the very least it is confusing...

**Comment by jeenbroekstra** [comment]
This has grown a bit stale I think @barthanssens - should we close this or do you want to take another stab at it?

**Comment by barthanssens** [comment]
Thanks for the reminder, I'll look into it

**Pull request closed without merging**